src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/IntervalTracker.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace NuGet.VisualStudio
+{
+    public class IntervalTracker
```

Comment:
This type seems to be useful in other non Visual Studio scenarios. Is it worth to be added to a NuGet Common , not NuGet.VisualStudio.Common?

Reply:
Stopwatch is not thread safe. If we need a thread safe version please find [here](http://joeduffyblog.com/2006/03/25/a-stopwatch-for-threads/) an idea.

Reply:
> This type seems to be useful in other non Visual Studio scenarios. Is it worth to be added to a NuGet Common , not NuGet.VisualStudio.Common?

We can move it later if we think it's necessary, but similar utilities already exist.
NuGet.VisualStudio.Common doesn't have the same semver contract concerns that NuGet.Common does given it's a widely used package. 

My personal rule is I won't define a reusable utility for 1 use case. 
A 2nd use cases gives us an idea of how a utility would look like, but more often the 3rd case is when we can feel more confident about the public API :)

Reply:
> Stopwatch is not thread safe. If we need a thread safe version please find here an idea.

That's a nice example, but for this use-case, it's wall clock time that we want to measure. 
We also know that all of the calls to the utility are sequential. 

Thread-safety is a very good point though, so I'll add a comment on the utility clarifying that.